kyo-data/shared/src/main/scala/kyo/KyoException.scala

```diff
@@ -0,0 +1,27 @@
+package kyo
+
+import kyo.*
+import kyo.Ansi.*
+import scala.util.control.NoStackTrace
+
+class KyoException private[kyo] (
+    message: Text = null,
+    cause: Text | Throwable = null
+)(using val frame: Frame) extends NoStackTrace:
+
+    override def getCause(): Throwable =
+        cause match
+            case cause: Throwable => cause
+            case _                => null
+
+    override def getMessage(): String =
+        val detail =
+            cause match
+                case _: Throwable           => Absent
+                case cause: Text @unchecked => Maybe(cause)
+
+        val msg = frame.render(("⚠️ KyoException".red.bold :: Maybe(message).toList ::: detail.toList).map(_.show)*)
+        s"\n$msg\n"
```

Comment:
Are the two `\n` really supposed to be there? Shouldn't formatting like this be potentially done by the users of `getMessage` instead of in `getMessage` itself?

Reply:
> provide better rendering in the console

Yeah, totally. But my point was that it seems like a concert of the code which does the rendering/printing/formatting of an exception to standard output. `getMessage` is generally implemented in a way which is independent/oblivious whether the target is standard output, HTML, JSON or whatever else.

It's a very small thing, so I don't mind at all, but it caught my attention, so I've mentioned it. Feel free to ignore :slightly_smiling_face: